src/sentry/models/authenticator.py

```diff
@@ -193,14 +193,18 @@ def requires_activation(self):
         """If the interface has an activation method that needs to be
         called this returns `True`.
         """
-        return self.activate.im_func is not AuthenticatorInterface.activate.im_func
+        return self.activate.__func__ is not six.get_unbound_function(
```

Comment:
Have you tried `six.get_method_function(self.activate) is not AuthenticatorInterface.activate`? I think then you might not have to mock __func__.

Reply:
https://github.com/benjaminp/six/blob/master/six.py#L573. It's __func__ in py3 and im_func in py2.

Reply:
mock functions don't have the attribute `__func__`

Reply:
Oh. You can use [autospeccing](https://docs.python.org/3/library/unittest.mock.html#auto-speccing) to make that available, pass [`autospec=True`](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.patch) to the patching. 

cc @wedamija; Autospeccing can avoid the silent failures with misspelling things like `assert_called_once`. I forget where, but this was a concern at some point.

Reply:
I also still think `six.get_method_function(self.activate) is not six.get_unbound_function(AuthenticatorInterface.activate)` is preferable.

Reply:
I found the answer on [SO answer](https://stackoverflow.com/a/31192136/90297):
```py
return self.activate.__code__ is not AuthenticatorInterface.activate.__code__
```

Reply:
Oh that's a decent solution

Reply:
I guess that works, but it feels weird to do that.

Reply:
tbh I'm fine with the solution I have right now, it looks good to me 🤷

Reply:
Your solution seems fine... is there any reason we can't just use `has_attr`?

Reply:
Nm, the interface already has it...

Reply:
Actually is there any reason we can't just do:
```
In [14]: class abc:
    ...:     def hi(self):
    ...:         pass
    ...:

In [15]: class abcd(abc):
    ...:     def hi(self):
    ...:         pass
    ...:

In [18]: class abcde(abc):
    ...:     pass
    ...:
    ...:

In [21]: abc.hi is abcd.hi
Out[21]: False

In [22]: abc.hi is abcde.hi
Out[22]: True
```

Reply:
Ahh, fails in python 2, dang. Ok, I think the unbound method way looks fine